src/js/card.js

```diff
@@ -0,0 +1,151 @@
+  constructor({ category, cover, words }, id) {
+    this.cardData = { category, cover, words }
+    this.id = id
+  }
+
+  render() {
+    const card = document.getElementById(this.id)
+    const quantityWords = card.querySelector('.info_block__quantity')
+
+    card.setAttribute('name', this.cardData.category)
+    card.classList.add('card')
+
+    const infoContainer = card.querySelector('.card__info_block')
+    const cover = card.querySelector('.card__cover')
+    const rotate = card.querySelector('.rotate')
+
+    if (card.querySelector('.info_block__subtitle')) {
+      const cardTitle = card.querySelector('.info_block__subtitle')
+      cardTitle.className = 'info_block__title'
+      cardTitle.innerText = this.cardData.category
+    }
+    if (card.querySelector('.info_block__title')) {
+      const cardTitle = card.querySelector('.info_block__title')
+      cardTitle.className = 'info_block__title'
+      cardTitle.innerText = this.cardData.category
+    }
+
+    cover.style.backgroundImage = `url('${this.cardData.cover}')`
+    if (quantityWords === null) {
+      const createQuantityWords = document.createElement('p')
+      createQuantityWords.classList.add('info_block__quantity')
+      infoContainer.appendChild(createQuantityWords)
+      createQuantityWords.innerText = this.cardData.words.length
+    }
+    rotate && infoContainer.removeChild(rotate)
+  }
+}
+
+class RenderSubCard {
+  constructor({ word, translation, image, audioSrc }, id) {
+    this.cardData = { word, translation, image, audioSrc }
+    this.id = id
+  }
+
+  render() {
+    const card = document.getElementById(this.id)
+
+    card.setAttribute('name', this.cardData.word)
+    card.classList.add('subCard')
+
+    const infoContainer = card.querySelector('.card__info_block')
+    const cover = card.querySelector('.card__cover')
+    const quantityWords = card.querySelector('.info_block__quantity')
+    const cardTitle = card.querySelector('.info_block__title')
+    const cardIndicator = card.querySelector('.info_block__indicator')
+    cardTitle.className = 'info_block__subtitle'
+
+    cardTitle.innerText = this.cardData.word
+    cover.style.backgroundImage = `url('${this.cardData.image}')`
+    const createRotate = document.createElement('img')
+    createRotate.classList.add('rotate')
+    createRotate.setAttribute('src', './assets/img/rotate.svg')
+    infoContainer.appendChild(createRotate)
+
+    quantityWords && infoContainer.removeChild(quantityWords)
+  }
+
+  rotate() {
+    const card = document.getElementById(this.id)
+    const rotateImg = card.querySelector('.rotate')
+    const title = card.querySelector('.info_block__subtitle')
+
+    function rotate(data) {
+      title.innerText = data
+      card.classList.add('_rotate')
+    }
+
+    function returns(data) {
+      title.innerText = data
+      card.classList.remove('_rotate')
+    }
+
+    rotateImg.addEventListener('click', () => {
+      rotate(this.cardData.translation)
```

Comment:
Looks too difficult. I think you don't need to create separate function. And please remove addEventListener after "using"

Reply:
+++